test/components/baseAxisTests.ts

```diff
@@ -208,4 +216,29 @@ describe("BaseAxis", () => {
     baseAxis = new Plottable.Axis.AbstractAxis(scale, "right");
     assert.equal((<any> baseAxis)._xAlignProportion, 0, "xAlignProportion defaults to 0 for right axis");
   });
+
+  it("tick labels do not overlap with each other", () => {
+    var SVG_WIDTH = 500;
+    var SVG_HEIGHT = 100;
+    var svg = generateSVG(SVG_WIDTH, SVG_HEIGHT);
+
+    var scale = new Plottable.Scale.Linear();
+    scale.domain([0, 1000000000000]);
+
+    var baseAxis = new Plottable.Axis.AbstractAxis(scale, "bottom");
+    baseAxis.renderTo(svg);
+
+    var visibleTickLabels = d3.selectAll(".x-axis .tick-label")
```

Comment:
I'd do
```Typescript
(<any> baseAxis)._element.selectAll(".tick-label")
```
since using `d3.selectAll` might grab labels left over elsewhere on the page from failed tests.